changelogs/unreleased/gh-10148-fix-small-crash-low-alloc-factor.md

```diff
@@ -0,0 +1,4 @@
+## bugfix/box
+
+- Fixed a crash when `slab_alloc_factor` is low and memory pressure is high
+  (gh-10148).
```

Comment:
Let's add a test for it here as well. I see that you already have a reproducer.

Reply:
Done.